lib/Onyx.js

```diff
@@ -732,10 +759,13 @@ function init({
 
     // Let Onyx know about which keys are safe to evict
     evictionAllowList = safeEvictionKeys;
-    addAllSafeEvictionKeysToRecentlyAccessedList();
 
-    // Initialize all of our keys with data provided
-    initializeWithDefaultKeyStates();
+    // Initialize all of our keys with data provided then give green light to any pending connections
+    Promise.all([
+        addAllSafeEvictionKeysToRecentlyAccessedList(),
+        initializeWithDefaultKeyStates()
+    ])
+        .then(deferredInit.resolve);
```

Comment:
It took me some time to wrap my head around this but this is a pretty cool trick we are doing with the deferred task stuff. I'm curious, how much better this would be compared to making `Onyx.init()` itself a promise and calling it before allowing any connections to happen? Does it take a significant amount of time to run this code?

Reply:
> Does it take a significant amount of time to run this code?

Before the change the initial `get` calls would start at 2ms past app launch and resolve at ~250ms 
After the change they would start at 212ms and resolve at 255ms 

So even though we delay things by 200ms they resolve at the same time, the speed up comes from the rest of the calls happening and in the end the last storage call is resolved 0.3-0.7 sec earlier

> I'm curious, how much better this would be compared to making `Onyx.init()` itself a promise and calling it before allowing any connections to happen?

Calling Onyx.init before any other Onyx calls would require many changes - every usage that is not a component based `withOnyx` can happen before Onyx.init is called, depending on script load order 
Even if we decide to do so it would not bring anything more to the table in terms of performance

Reply:
>the speed up comes from the rest of the calls happening

Hmm which calls? Sorry not sure I understand. Could you maybe say it another way?

>in the end the last storage call is resolved 0.3-0.7 sec earlier

It looks like iOS experienced almost no changes (and in some cases got worse)? Does that sound right? If so, can you account for the inconsistency there?

>Calling Onyx.init before any other Onyx calls would require many changes - every usage that is not a component based withOnyx can happen before Onyx.init is called, depending on script load order

👍